core/src/memory/resources/terminal/malloc_resource.cpp

```diff
@@ -0,0 +1,33 @@
+#include "vecmem/memory/resources/memory_resource.hpp"
+#include "vecmem/memory/resources/terminal/malloc_resource.hpp"
+
+#include <memory>
+
+namespace vecmem::resource {
+    void * malloc_resource::do_allocate(
+        std::size_t bytes,
+        std::size_t
+    ) {
+        return malloc(bytes);
+    }
+
+    void malloc_resource::do_deallocate(
+        void * p,
+        std::size_t,
+        std::size_t
+    ) {
+        free(p);
+    }
+
+    bool malloc_resource::do_is_equal(
+        const memory_resource &
+    ) const noexcept {
+        /*
+         * All malloc resources are equal to each other, because they have no
+         * internal state. Of course they have a shared underlying state in the
+         * form of the underlying C library memory manager, but that is not
+         * relevant for us.
+         */
+        return true;
```

Comment:
:smile: Now that I pondered about this a bit more... I think this should be implemented as:

```c++
bool host_memory::do_is_equal( const memory_resource& other ) const noexcept {
   const host_memory* otherPtr = dynamic_cast< const host_memory* >( &other );
   return ( otherPtr != nullptr );
}
```

Since STL may very well be comparing this object instance to some GPU memory resource...

The GPU resources would then not only make sure that the "other" object is the same type as they are but that it is actually also talking to the same GPU as they are.